src/filtering/convert.rs

```diff
@@ -31,11 +33,119 @@ pub async fn convert(cnf: PathBuf) -> Result<SparseMarathonfile> {
     Ok(filtering_configuration)
 }
 
+pub async fn convert_xctestplan(
+    cnf: PathBuf,
+    target_name: Option<String>,
+) -> Result<SparseMarathonfile> {
+    let content = fs::read_to_string(&cnf)
+        .await
+        .map_err(|error| InputError::OpenFileFailure {
+            path: cnf.clone(),
+            error,
+        })?;
+
+    let xctestplan: xctestplan::SparseTestPlan = serde_json::from_str(&content)?;
+    let targets = xctestplan.test_targets;
+    let target = match target_name {
+        Some(target_name) => targets
+            .iter()
+            .find(|x| x.target.name == target_name)
+            .ok_or(InputError::XctestplanMissingTargets)?,
+        None => targets
+            .first()
+            .ok_or(InputError::XctestplanMissingTargets)?,
+    };
+
+    let allowlist = target
+        .selected_tests
+        .as_ref()
+        .map(|x| xctestplan_ids_to_filter(x));
+    let blocklist = target
+        .skipped_tests
+        .as_ref()
+        .map(|x| xctestplan_ids_to_filter(x));
+
+    let filtering_configuration = FilteringConfiguration {
+        allowlist: allowlist.map(|x| vec![x]),
+        blocklist: blocklist.map(|x| vec![x]),
+    };
+    let marathonfile = SparseMarathonfile {
+        filtering_configuration,
+    };
+
+    Ok(marathonfile)
+}
+
+//Identifiers contain a mix of class names and class name with method signature
+//Sometimes you can see separator \/ and sometimes / for the class and method
+//Also sometime ending () are present for method filtering
+fn xctestplan_ids_to_filter(ids: &[String]) -> Filter {
+    let mut class_names: Vec<String> = vec![];
+    let mut simple_test_names: Vec<String> = vec![];
+
+    ids.iter().for_each(|id| {
+        if id.contains("/") || id.contains("(") {
+            let marathon_id = id.replace("\\/", "#").replace("/", "#").replace("()", "");
+            simple_test_names.push(marathon_id);
+        } else {
+            class_names.push(id.clone());
+        }
+    });
+
+    if !class_names.is_empty() && !simple_test_names.is_empty() {
+        //Need to use composition since filtering is done via class names and also using methods
+        let class_name_filter = Filter {
+            mtype: "simple-class-name".into(),
+            values: Some(class_names),
+            op: None,
+            file: None,
+            regex: None,
+            filters: None,
+        };
+
+        let simple_qualified_test_name_filter = Filter {
+            mtype: "simple-test-name".into(),
```

Comment:
Is that possible to use only test name? What about coincidence of the same test name in different classes? Or xctestplan allows this trick and it is a user responsibility?